design/ceph-csi-driver.md

```diff
@@ -54,6 +56,19 @@ If CSI drivers are already successfully deployed, Rook should help monitor the S
 
 ### Rook initiated CSI drivers deployment
 
-If Rook is instructed to deploy CSI drivers and other external controllers, Rook Operator should create all the driver and daemon deployments as well.
+If Rook is instructed to deploy CSI drivers and other external controllers, Rook Operator should create all the driver and daemon deployments as well. Since there are breaking changes in CSI 1.0 spec and Kubernetes 1.13, it makes sense to focus on newer CSI specs and only on latest Kubernetes releases (i.e. 1.13 and beyond).
+
+The changes to Rook include the following:
+
+- Create a new API object for CSI driver deployment. Such object should inform Rook Ceph Operator which driver (rbd or Ceph FS, or both) to start, in which namespace to run the CSI drivers, and which Ceph cluster to get configuration infomation to create Storage Class and Secret.
+- Operator starts CSI Drivers and Controllers according to the new API spec.
+
+In the near term (Kubernetes 1.13 and maybe 1.14 timeframe), we adopt the following approaches, allowing us to use latest CSI driver image and runtime configuration:
+- Use Golang template to materialize Kubernetes and Rook API objects.
+- Consume the same Container images as provided by Ceph CSI. 
+
+In the long run, when CSI spec and drivers stablize, we will switch to the following:
+
+- Vendor in Ceph CSI drivers and create CSI driver binaries in Rook Ceph container image, rather than directly consuming ceph-csi images.
```

Comment:
I don't see how vendoring in the csi drivers will allow better supportability than consuming the csi driver images that are built separately. Seems like ceph-csi will need to support the templating or other mechanism to support multiple versions of k8s anyway. And either way, rook would run the integration test on the matrix of versions that are supported.